src/net_processing.cpp

```diff
@@ -4210,6 +4219,23 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         m_txrequest.ReceivedResponse(pfrom.GetId(), txid);
         if (tx.HasWitness()) m_txrequest.ReceivedResponse(pfrom.GetId(), wtxid);
 
+        if (UseSensitiveRelay() && m_mempool.RemoveUnbroadcastTx(txid)) {
+            LogPrintLevel( /* Continued */
+                BCLog::SENSITIVE_RELAY,
+                BCLog::Level::Info,
+                "Received own transaction (txid=%s) from the network from peer=%d%s; "
+                "stopping sensitive broadcast and broadcasting it to everybody as if it is not our transaction\n",
```

Comment:
This chain of peers is desired / demanded to consider the transaction successfully propagated. Lets say we are connected to 30 peers (some inbound, some outbound). If somebody of them gets the transaction (from somebody else, not from us) he will tell us about it due to the normal flooding mechanism because he will not be aware we have it. If all our 30 peers stay silent about the transaction, it means that they did not get it which is a sign it propagated only to some part of the network or did not propagate at all.

My understanding is that normally it takes a few seconds for a transaction to reach _every node_ in the network due to the flooding mechanism.